receiver/azuremonitorreceiver/scraper.go

```diff
@@ -587,3 +664,16 @@ func filterResourceTags(tagFilterList map[string]struct{}, resourceTags map[stri
 
 	return includedTags
 }
+
+func copyResource(original *armresources.GenericResourceExpanded) (*armresources.GenericResourceExpanded, error) {
+	hackedResourceStr, err := original.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	var newResource armresources.GenericResourceExpanded
+	err = newResource.UnmarshalJSON(hackedResourceStr)
+	if err != nil {
+		return nil, err
+	}
+	return &newResource, nil
```

Comment:
This feels rather heavy handed in order to perform a deep clone, is there a way to do this without constructing a json object?